contracts/volt/VoltV2.sol

```diff
@@ -0,0 +1,528 @@
+    mapping(address => mapping(uint32 => Checkpoint)) public checkpoints;
+
+    /// @notice The number of checkpoints for each account
+    mapping(address => uint32) public numCheckpoints;
+
+    /// @notice The EIP-712 typehash for the contract's domain
+    bytes32 public constant DOMAIN_TYPEHASH =
+        keccak256(
+            "EIP712Domain(string name,uint256 chainId,address verifyingContract)"
+        );
+
+    /// @notice The EIP-712 typehash for the delegation struct used by the contract
+    bytes32 public constant DELEGATION_TYPEHASH =
+        keccak256("Delegation(address delegatee,uint256 nonce,uint256 expiry)");
+
+    /// @notice The EIP-712 typehash for the permit struct used by the contract
+    bytes32 public constant PERMIT_TYPEHASH =
+        keccak256(
+            "Permit(address owner,address spender,uint256 value,uint256 nonce,uint256 deadline)"
+        );
+
+    /// @notice A record of states for signing / validating signatures
+    mapping(address => uint256) public nonces;
+
+    /// @notice An event thats emitted when an account changes its delegate
+    event DelegateChanged(
+        address indexed delegator,
+        address indexed fromDelegate,
+        address indexed toDelegate
+    );
+
+    /// @notice An event thats emitted when a delegate account's vote balance changes
+    event DelegateVotesChanged(
+        address indexed delegate,
+        uint256 previousBalance,
+        uint256 newBalance
+    );
+
+    /// @notice The standard EIP-20 transfer event
+    event Transfer(address indexed from, address indexed to, uint256 amount);
+
+    /// @notice The standard EIP-20 approval event
+    event Approval(
+        address indexed owner,
+        address indexed spender,
+        uint256 amount
+    );
+
+    constructor(address core) CoreRef(core) {}
+
+    /// @notice Mint new tokens
+    /// @param dst The address of the destination account
+    /// @param rawAmount The number of tokens to be minted
+    function mint(address dst, uint256 rawAmount)
+        external
+        hasAnyOfTwoRoles(TribeRoles.GOVERNOR, TribeRoles.MINTER)
+    {
+        require(dst != address(0), "Volt: cannot transfer to the zero address");
+
+        // mint the amount
+        uint96 amount = safe96(rawAmount, "Volt: amount exceeds 96 bits");
+        uint96 safeSupply = safe96(
+            totalSupply,
+            "Volt: totalSupply exceeds 96 bits"
+        );
+        totalSupply = add96(
+            safeSupply,
+            amount,
+            "Volt: totalSupply exceeds 96 bits"
+        );
+
+        // transfer the amount to the recipient
+        balances[dst] = add96(
+            balances[dst],
+            amount,
+            "Volt: transfer amount overflows"
+        );
+        emit Transfer(address(0), dst, amount);
+
+        // move delegates
+        _moveDelegates(address(0), delegates[dst], amount);
+    }
+
+    function burn(address dst, uint256 rawAmount)
```

Comment:
let's delete this function.

Reply:
we need a burn function where the owner can burn their own tokens, not others tokens. then we need a burnFrom function that allows an address to burn someone else's token's provided they have the approval to spend those tokens.